hack/generator/pkg/codegen/storage/type_converter.go

```diff
@@ -91,6 +155,13 @@ func (t *TypeConverter) convertObjectType(
 	}
 
 	objectType := astmodel.NewObjectType().WithProperties(properties...)
+
+	if converterContext.isSpec {
+		// Inject OriginalVersion property
+		originalVersion := astmodel.NewPropertyDefinition("OriginalVersion", "original-version", astmodel.StringType)
```

Comment:
Same comment as in the golden file, I think this should be `originalVersion` not `original-version`